bin/tasks/gitInfo.js

```diff
@@ -72,11 +73,27 @@ export const setGitInfo = async (ctx, task) => {
     const mostRecentAncestor = await ctx.client.runQuery(TesterLastBuildQuery, { commit, branch });
     if (mostRecentAncestor && ['PASSED', 'ACCEPTED'].includes(mostRecentAncestor.status)) {
       ctx.skip = true;
-      return transitionTo(skippedRebuild, true)(ctx, task);
+      transitionTo(skippedRebuild, true)(ctx, task);
+      return;
     }
   }
 
-  return transitionTo(success, true)(ctx, task);
+  // Retrieve a list of changed file paths since the baseline commit(s), which will be used to
+  // determine affected story files later.
+  // In the unlikely scenario that this list is empty (and not a rebuild), we can skip the build
+  // completely, since we know for certain it wouldn't have any effect.
+  if (baselineCommits.length && matchesBranch(ctx.options.onlyChanged)) {
+    const results = await Promise.all(baselineCommits.map((c) => getChangedFiles(c)));
+    ctx.git.changedFiles = [...new Set(results.flat())].map((f) => `./${f}`);
+    ctx.log.debug(`Found changedFiles:\n${ctx.git.changedFiles.map((f) => `  ${f}`).join('\n')}`);
+    if (ctx.git.changedFiles.length === 0) {
+      ctx.skip = true;
+      transitionTo(skippedRebuild, true)(ctx, task);
```

Comment:
Should this transition to `skippedForCommit`? What does `skippedRebuild` do?

In this case you would want to mark the commit as successful, ie call the `skipBuild` mutation. Maybe it is better to let this fall out naturally from calling `getDependentStoryFiles` with an empty list of files?